hack/validate-generate

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail -x
+
+iidfile=$(mktemp -t docker-iidfile.XXXXXXXXXX)
+docker build --build-arg GOGO_VERSION=eda389564aa83cc6949f47e0524c47cc2e4aaa23 --iidfile $iidfile -f ./hack/dockerfiles/generate.Dockerfile --force-rm .
+iid=$(cat $iidfile)
+diffs="$(docker run $iid git status --porcelain -- **/*.pb.go 2>/dev/null)"
```

Comment:
How about putting the generate and `git status` both in Dockerfile but to different stages. For example `generate` can create a tarball from scratch so it would also work with the local exporter in the future.

Reply:
Do you mean for `generate` to create `FROM scratch` with the unpacked files (usable with `docker export`) rather than the tarball? I'll try that it might be neat.

Reply:
Unfortunately the result is to add `.dockerenv` file and `sys`, `proc` & `dev` subdirectories to the source tree, which isn't exactly what we want!

I've managed to make it work by populating the files in `/generated-files` and using `tar`'s `--strip-components` option which (from the manpages) works on both Mac (untested) and Linux (tested).